### ISSUE TITLE
Dallas temperature sensor request

### ISSUE THREAD

**Issue by Vado75** [nthnn/PH4502C-Sensor#5]
Hi, thanks for the ph library. Would you be so kind as to add functionality for the Dallas external temp sensor to compensate ph from temp please? I want to use the ph probe in the pipe and measure the temperature using dallas. Thank you.

**Comment by nthnn** [comment]
Hi! The PH4502C sensor typically have a temperature output pin on breakout boards. Can you please clarify what do you mean by "Dallas external temperature sensor"? Perhaps you mean DS18B20? (I found it on a quick Google search.)

**Comment by artech-technology** [comment]
yes, the temperature sensor is on the PCB, but if you want to compensate the ph of the measured solution, you need the temperature of the solution. In fact, measuring the temperature on the board is useless for this reason. That's why I wanted to ask you for a variation of the measurement with an external temperature sensor - the ideal candidate is the DS18B20 - it has a waterproof case, it is easily implemented in the measured pipe together with the PH probe.
Can you please share the part of the code with temperature compensation with respect to PH?
Note, I can use OneWire to read the temperature from Dallas if you extend your program for example ph4502.read_ph_level_single(temperature)
thank you

**Comment by nthnn** [comment]
Ohhh, right! I now understand what do you mean earlier. However, I'm afraid I currently don't have an allocated budget to buy my own DS18B20 for testing, experimentations, and validation. Additionally, I just borrowed the PH4502C sensor while developing this library.

So unfortunately, it may take some time to do that update. But thank you for that great idea!

**Comment by artech-technology** [comment]
ok i understand. And does your library compensate for PH based on temperature?

**Comment by nthnn** [comment]
Currently, no. It does not compensate pH calculations based on temperature.

**Comment by artech-technology** [comment]
understand, temperature is only for information, not for compesantion... thank you, have a nice day

**Comment by nthnn** [comment]
Have a nice day!
